dstep/translator/Translator.d

```diff
@@ -69,48 +70,62 @@ class Translator
         write(outputFile, translateToString());
     }
 
+    static size_t countTrailingEndls(string a) 
+    {
+        size_t itr = a.length;
+
+        while (itr != 0 && a[itr - 1] == '\n')
+            --itr;
+                
+        return a.length - itr;
+    }
+
+    static size_t countLeadingEndls(string a)
+    {
+        size_t count = 0;
+
+        while (count < a.length && a[count] == '\n')
+            ++count;
+
+        return count;
+    }
+
+    static void endlAwareAppend(ref Appender!string a, string b)
+    {
+        import std.algorithm.comparison;
+
+        size_t aTrailing = countTrailingEndls(a.data);
+        size_t bLeading = countLeadingEndls(b);
+        size_t bTrailing = countTrailingEndls(b);
+
+        size_t num = min(bLeading + aTrailing, 2);
+        a.put("\n".replicate(num < aTrailing ? 0 : num - aTrailing));
+        a.put(b[bLeading..$-bTrailing]);
+        a.put("\n".replicate(min(bTrailing, 2)));
+    }
```

Comment:
Then `endlAwareAppend` tries to join two strings together (`Appender` and `string` actually) in the way there is never more than two `\n` between them. It should collaborate with "translate" methods. 

Translate method can return different combinations of end-lines e.g.:

- `content\n`
- `\ncontent\n`
- `content\n\n`

Signalling that the item should be followed by `\n`, preceded by `\n` and  and followed by `\n` and followed by double `\n` respectively. Then `endlAwareAppend` can intelligently merge two items by taking the sum of `\n` but not more than 2.

For example, let `++` be `endlAwareAppend` equivalent:

- `"\ncontent\n" ++ "\ncontent\n" === "\ncontent\n\ncontent\n"`
- `"enum FOO = 1;\n" ++ "enum BAR = 2;\n" === "enum FOO = 1;\nenum BAR = 2;\n"`
- `"enum FOO = 1;\n" ++ "\nclass A { }; \n" === "enum FOO = 1;\n\nclass A { };\n"`
- `"enum WANT_BLANK_AFTER_THIS = 0;\n\n" ++ "\nclass WANT_BLANK_BEFORE_AND_AFTER_THIS { }\n\n" === "enum WANT_BLANK_AFTER_THIS = 0;\n\nclass WANT_BLANK_BEFORE_AND_AFTER_THIS { }\n\n"`

Saying it another way you precede item by `\n` if you want blank before the item and follow the item by double `\n` if you want a blank newline after item and the `endlAwareAppend` takes care of joining it so that there is no more than one blank line.